lib/Cake/Core/App.php

```diff
@@ -889,6 +889,36 @@ protected static function _packageFormat() {
 	}
 
 /**
+ * Increases the PHP "memory_limit" ini setting by the specified amount
+ * in megabytes  
+ *  
+ * @param string $additionalMb Number in megabytes
+ * @return void
+ */
+	public static function increasePhpMemoryLimit($additionalMb) {
+		$limit = ini_get("memory_limit");
+		if (!is_string($limit) || !strlen($limit)) {
+			return;
+		}
+
+		$limit = trim($limit);
+		$units = strtoupper(substr($limit, -1));
+		$current = substr($limit, 0, strlen($limit) - 1);
+		if ($units === "K") {
```

Comment:
You can use `CakeNumber::fromReadableSize()` and `CakeNumber::toReadableSizeable()` instead of all this unit conversion code.